matrix/stdlib_imports.hpp

```diff
@@ -13,6 +13,14 @@
 #include <cstdlib>
 #include <cinttypes>
 
+#ifndef M_PI
+#define M_PI	3.14159265358979323846
```

Comment:
Can we avoid defining our own PI? It's not standard, but I believe all of our platforms support _USE_MATH_DEFINES.

Reply:
That was exactly my goal and when I run the tests on my computer it all works but on CI not. So I tried
```
#define _USE_MATH_DEFINES
#undef __STRICT_ANSI__
#include <cmath>
```
but CI still didn't find pi... So if you have any clue if we can change the CI call such that it works.

Reply:
Agree that rediscovering PI is not ideal and may be another good change, but isn't directly related to the PR's primary objective. To keep from blocking maybe a new branch?